yt_dlp/extractor/nhk.py

```diff
@@ -684,67 +684,59 @@ class NhkRadiruIE(InfoExtractor):
         detail_url = try_call(
             lambda: self._API_URL_TMPL.format(area=area, service=service, dateid=aa_vinfo[3]))
-
         if not detail_url:
             return {}
-        response = self._download_json(detail_url, episode_id, 'Downloading extended metadata', fatal=False, expected_status=400)
 
-        error = response.get('error')
-        if error:
-            self.report_warning(f'Failed to get extended metadata. API returned Error {error.get("code")}: {error.get("message")}')
+        response = self._download_json(
+            detail_url, episode_id, 'Downloading extended metadata',
+            'Failed to download extended metadata', fatal=False, expected_status=400)
+        if not response:
             return {}
-        if response.get('list') is None:
-            self.report_warning('Failed to get extended metadata. API returned empty list.')
+
+        if error := traverse_obj(response, ('error', {dict})):
+            self.report_warning(
+                'Failed to get extended metadata. API returned '
+                f'Error {join_nonempty("code", "message", from_dict=error, delim=": ")}')
             return {}
 
         full_meta = traverse_obj(response, ('list', service, 0, {dict}))
+        if not full_meta:
+            self.report_warning('Failed to get extended metadata. API returned empty list.')
+            return {}
 
-        station = ' '.join(traverse_obj(full_meta, (('service', 'area'), 'name', {str})))
-        description = join_nonempty('subtitle', 'content', 'act', 'music', delim='\n\n', from_dict=full_meta)
-
-        thumbnails = []
-        for img_id, img_info in full_meta.get('images').items():
-            if img_info.get('url') != '':  # can happen
-                thumb_dict = {
-                    'id': img_id,
-                    **traverse_obj(img_info, {
-                        'url': 'url',
-                        'width': ('width', {int_or_none}),
-                        'height': ('height', {int_or_none}),
-                    }),
-                }
-                if img_id.startswith('thumbnail'):
-                    thumb_dict['preference'] = 1
-                elif img_id.startswith('logo'):
-                    thumb_dict['preference'] = -2
-                thumbnails.append(thumb_dict)
+        station = ' '.join(traverse_obj(full_meta, (('service', 'area'), 'name', {str}))) or None
+        thumbnails = [{
+            'id': str(id_),
+            'preference': 1 if id_.startswith('thumbnail') else -2 if id_.startswith('logo') else -1,
+            **traverse_obj(thumb, {
+                'url': 'url',
+                'width': ('width', {int_or_none}),
+                'height': ('height', {int_or_none}),
+            }),
+        } for id_, thumb in traverse_obj(full_meta, ('images', {dict.items}, lambda _, v: v[1]['url']))]
 
-        return {
+        return filter_dict({
```

Comment:
need to filter out fields with `None` values or else we overwrite the fallback values when we unpack this

Reply:
:+1:, I knew that would happen at the time but didn't know how to handle it, thanks